modAionImpl/src/org/aion/zero/impl/sync/TaskImportBlocks.java

```diff
@@ -64,204 +80,622 @@
 
     private final Logger log;
 
+    private SortedSet<Long> baseList;
+    private PeerState state;
+
     TaskImportBlocks(
             final AionBlockchainImpl _chain,
             final AtomicBoolean _start,
             final SyncStatics _statis,
-            final BlockingQueue<BlocksWrapper> downloadedBlocks,
-            final Map<ByteArrayWrapper, Object> importedBlockHashes,
-            final Map<Integer, PeerState> peerStates,
-            final Logger log) {
+            final BlockingQueue<BlocksWrapper> _downloadedBlocks,
+            final Map<ByteArrayWrapper, Object> _importedBlockHashes,
+            final Map<Integer, PeerState> _peerStates,
+            final Logger _log) {
         this.chain = _chain;
         this.start = _start;
         this.statis = _statis;
-        this.downloadedBlocks = downloadedBlocks;
-        this.importedBlockHashes = importedBlockHashes;
-        this.peerStates = peerStates;
-        this.log = log;
+        this.downloadedBlocks = _downloadedBlocks;
+        this.importedBlockHashes = _importedBlockHashes;
+        this.peerStates = _peerStates;
+        this.log = _log;
+        this.baseList = new TreeSet<>();
+        this.state = new PeerState(NORMAL, 0L);
     }
 
     @Override
     public void run() {
         Thread.currentThread().setPriority(Thread.MAX_PRIORITY);
         while (start.get()) {
-
             BlocksWrapper bw;
             try {
                 bw = downloadedBlocks.take();
             } catch (InterruptedException ex) {
                 return;
```

Comment:
Add error log in this exception.

Reply:
Please check if you agree with my change. I am logging the error only when shutdown was not called since it is to be expected that this will get interrupted at shutdown.